File: app/src/main/java/com/rolandvitezhu/todocloud/ui/activity/main/viewmodel/PredefinedListsViewModel.java
package com.rolandvitezhu.todocloud.ui.activity.main.viewmodel;

import com.rolandvitezhu.todocloud.data.PredefinedList;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class PredefinedListsViewModel extends ViewModel {

  private MutableLiveData<List<PredefinedList>> predefinedLists;
  private PredefinedList predefinedList;

  public LiveData<List<PredefinedList>> getPredefinedLists() {
    if (predefinedLists == null)
      predefinedLists = new MutableLiveData<>();

    return predefinedLists;
  }

  public void setPredefinedLists(List<PredefinedList> predefinedLists) {
    if (this.predefinedLists == null)
      this.predefinedLists = new MutableLiveData<>();

    this.predefinedLists.setValue(predefinedLists);
  }

  public PredefinedList getPredefinedList() {
    return predefinedList;
  }

  public void setPredefinedList(PredefinedList predefinedList) {
    this.predefinedList = predefinedList;
  }
}

File: app/src/main/java/com/rolandvitezhu/todocloud/ui/activity/main/viewmodel/PredefinedListsViewModel.kt
package com.rolandvitezhu.todocloud.ui.activity.main.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.rolandvitezhu.todocloud.R
import com.rolandvitezhu.todocloud.app.AppController.Companion.appContext
import com.rolandvitezhu.todocloud.app.AppController.Companion.instance
import com.rolandvitezhu.todocloud.data.PredefinedList
import com.rolandvitezhu.todocloud.database.TodoCloudDatabaseDao
import java.util.*
import javax.inject.Inject

class PredefinedListsViewModel : ViewModel() {

    @Inject
    lateinit var todoCloudDatabaseDao: TodoCloudDatabaseDao

    private val _predefinedLists = MutableLiveData<List<PredefinedList>>()
    val predefinedLists: LiveData<List<PredefinedList>>
        get() = _predefinedLists
    var predefinedList = PredefinedList()

    /**
     * Set the list of predefined lists.
     */
    suspend fun updatePredefinedListsViewModel() {
        val todayPredefinedListWhere: String = todoCloudDatabaseDao.prepareTodayPredefinedListWhere()
        val next7DaysPredefinedListWhere: String = todoCloudDatabaseDao.prepareNext7DaysPredefinedListWhere()
        val allPredefinedListWhere: String = todoCloudDatabaseDao.prepareAllPredefinedListWhere()
        val completedPredefinedListWhere: String = todoCloudDatabaseDao.prepareCompletedPredefinedListWhere()

        val predefinedLists = ArrayList<PredefinedList>()

        predefinedLists.add(
                PredefinedList(appContext.getString(R.string.all_today), todayPredefinedListWhere))
        predefinedLists.add(
                PredefinedList(appContext.getString(R.string.all_next7days), next7DaysPredefinedListWhere))
        predefinedLists.add(
                PredefinedList(appContext.getString(R.string.all_all), allPredefinedListWhere))
        predefinedLists.add(
                PredefinedList(appContext.getString(R.string.all_completed), completedPredefinedListWhere))

        _predefinedLists.value = predefinedLists
    }

    init {
        instance?.appComponent?.inject(this)
    }
}